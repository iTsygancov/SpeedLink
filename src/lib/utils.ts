import { type ClassValue, clsx } from "clsx";
import { twMerge } from "tailwind-merge";

export function cn(...inputs: ClassValue[]) {
  return twMerge(clsx(inputs));
}

export function generateAlphanumericKeysArray() {
  const keysArray = [];

  // Generate lowercase alphabetic keys
  for (let i = 65; i <= 90; i++) {
    keysArray.push(String.fromCharCode(i));
  }

  // Generate numeric keys
  for (let i = 48; i <= 57; i++) {
    keysArray.push(String.fromCharCode(i));
  }

  return keysArray;
}
