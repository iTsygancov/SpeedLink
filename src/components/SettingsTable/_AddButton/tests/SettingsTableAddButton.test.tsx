import SettingsTableAddButton, {
  SettingsTableEditCellProps
} from "../SettingsTableAddButton";
import { expectToMatchSnapshot, renderTree } from "@/test/utils";
import { describe, it, vi } from "vitest";

const settingsTableAddButtonProps: SettingsTableEditCellProps = {
  disabled: false,
  handleAddNewShortCut: vi.fn()
};

const testComponentWithProps = (props: SettingsTableEditCellProps) => {
  expectToMatchSnapshot(renderTree(<SettingsTableAddButton {...props} />));
};

describe("SettingsTableAddButton", () => {
  it("should render normally", () => {
    testComponentWithProps(settingsTableAddButtonProps);
  });

  it("should render with disabled button", () => {
    testComponentWithProps({
      ...settingsTableAddButtonProps,
      disabled: true
    });
  });
});
