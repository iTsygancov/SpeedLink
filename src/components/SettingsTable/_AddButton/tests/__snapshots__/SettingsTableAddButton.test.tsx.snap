// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SettingsTableAddButton > should render normally 1`] = `
<div
  className="mt-6 text-left"
>
  <button
    className="active:shadow-inner hover:shadow-sm justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 flex items-center gap-2"
    disabled={false}
    onClick={[MockFunction spy]}
  >
    <svg
      className="lucide lucide-plus"
      fill="none"
      height={24}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M5 12h14"
      />
      <path
        d="M12 5v14"
      />
    </svg>
    Add new shortcut
  </button>
</div>
`;

exports[`SettingsTableAddButton > should render with disabled button 1`] = `
<div
  className="mt-6 text-left"
>
  <button
    className="active:shadow-inner hover:shadow-sm justify-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 flex items-center gap-2"
    disabled={true}
    onClick={[MockFunction spy]}
  >
    <svg
      className="lucide lucide-plus"
      fill="none"
      height={24}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M5 12h14"
      />
      <path
        d="M12 5v14"
      />
    </svg>
    Add new shortcut
  </button>
</div>
`;
